registry/Scripts/pre-reqs.ps1

```diff
@@ -65,6 +68,173 @@ function New-ResourceGroup (
     }
 }
 
+<#
+.Synopsis
+    Create a DNS zone under given resource group
+
+.Description
+    Create a DNS zone under given resource group
+    It will skip creation of DNS zone resource if already present.
+    Return storage account details.
+
+.Parameter ResourceGroupName
+    DNS zone's resource group.
+
+.Parameter Name
+    DNS zone name.
+
+.Example
+   New-DnsZone -ResourceGroupName "resourcegroupname"
+                      -Name "sample.com"
+#>
+function New-DnsZone (
+    [Parameter(Mandatory = $true, HelpMessage = "DNS zone's resource group.")]
+    [string] $ResourceGroupName,
+    [Parameter(Mandatory = $true, HelpMessage = "DNS zone name.")]
+    [string] $Name
+)
+{
+    # Create storage account
+    
+    Write-Host "Check if DNS Zone ($Name) already exists"
+    $ErrorActionPreference = "SilentlyContinue";
+    $dnsZoneDetails = Get-AzureRmDnsZone -Name $Name `
+                                         -ResourceGroupName $ResourceGroupName `
+                                         -ErrorVariable dnsZoneExistError
+    $ErrorActionPreference = "Continue"; #Turning errors back on
+    if (-not $dnsZoneDetails) {
+        Write-Host "DNS zone does not exists. Creating a new DNS zone ($Name) under resource group ($ResourceGroupName)." 
+        New-AzureRmDnsZone -Name $Name -ResourceGroupName $ResourceGroupName | Out-Null
+        #$ErrorActionPreference = "SilentlyContinue";
+        $newDNSZoneDetails = Get-AzureRmDnsZone -Name $Name -ResourceGroupName $ResourceGroupName `
+                                                       -ErrorVariable newDNSZoneCreateError
+        #$ErrorActionPreference = "Continue"; #Turning errors back on
+        if  (-not $newDNSZoneDetails) {
+            throw "Creation a new DNS zone ($Name) under resource group ($ResourceGroupName) failed. Check if specified DNS zone already exists under a subscription." 
```

Comment:
`creation of` or `creating a`
capital Z, `Zone`
`under a subscription` is not clear

Reply:
updated to Check if specified DNS zone with same name already exists under the selected subscription